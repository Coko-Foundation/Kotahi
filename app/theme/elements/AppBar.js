import { css } from 'styled-components'
import { th } from '@pubsweet/ui-toolkit'

/*
  To disable underline from Logo
*/

export default {
  Root: css`
    box-shadow: 0 0 1px ${th('colorPrimary')};
    margin-bottom: 1px;
  `,
  LogoLink: css`
    &:hover:before {
      visibility: hidden;
    }
  `,
}
