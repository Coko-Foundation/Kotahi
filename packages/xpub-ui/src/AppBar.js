import React from 'react'
import { Link } from 'react-router'
import classnames from 'classnames'
import classes from './AppBar.local.css'

const AppBar = ({ brandLink, brandName, loginLink, logoutLink, userName }) => (
  <div className={classes.root}>
    <Link to={brandLink || '/'}
          className={classes.link}>{brandName}</Link>

    <div>
      {userName && (
        <span className={classes.item}>{userName}</span>
      )}

      {userName ? (
        <Link to={logoutLink}
              className={classnames(classes.item, classes.link)}>logout</Link>
      ) : (
        <Link to={loginLink}
              className={classnames(classes.item, classes.link)}>login</Link>
      )}
    </div>
  </div>
)

export default AppBar
