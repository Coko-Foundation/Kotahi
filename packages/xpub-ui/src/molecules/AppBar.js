import React from 'react'
import { Link } from 'react-router-dom'
import classnames from 'classnames'
import classes from './AppBar.local.scss'
import Icon from '../atoms/Icon'

const AppBar = ({ brandLink, brandName, loginLink, logoutLink, userName }) => (
  <div className={classes.root}>
    <Link
      to={brandLink || '/'}
      className={classnames(classes.link, classes.logo)}
    >
      {brandName}
    </Link>

    <div>
      {userName && (
        <span className={classes.item}>
          <Icon size={16}>user</Icon>
          <span className={classes.username}>{userName}</span>
        </span>
      )}

      {userName ? (
        <Link
          to={logoutLink}
          className={classnames(classes.item, classes.link)}
        >
          logout
        </Link>
      ) : (
        <Link to={loginLink} className={classnames(classes.item, classes.link)}>
          login
        </Link>
      )}
    </div>
  </div>
)

export default AppBar
