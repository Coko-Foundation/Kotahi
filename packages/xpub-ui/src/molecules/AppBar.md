The app bar appears at the top of every page of the application.

It displays the name of the application (as a link to the home page), the username of the current user, and a link to sign out. 

```js
<AppBar
  brandName="xpub"
  loginLink="/login"
  logoutLink="/logout"
  userName="foo"/>
```

When the user is not signed in, only the login link is displayed.

```js
<AppBar
  brandName="xpub"
  loginLink="/login"
  logoutLink="/logout"/>
```
