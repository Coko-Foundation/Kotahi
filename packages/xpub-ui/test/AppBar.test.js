import React from 'react'
import { clone } from 'lodash'
import { shallow } from 'enzyme'
import { Link } from 'react-router'
import renderer from 'react-test-renderer'

import AppBar from '../src/molecules/AppBar'

const props = {
  brandLink: 'some link',
  brandName: 'some brand',
  loginLink: 'login link',
  logoutLink: 'logout link',
  userName: 'some name',
}

const wrapper = shallow(<AppBar {...props} />)

describe('AppBar', () => {
  test('Snapshot', () => {
    const tree = renderer.create(
      <AppBar {...props} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Should render correctly', () => {
    expect(wrapper.is('div'))
    expect(wrapper.children()).toHaveLength(2)

    const brand = wrapper.childAt(0)
    expect(brand.is(Link)).toBeTruthy()
    expect(brand.children()).toHaveLength(1)

    const rightArea = wrapper.childAt(1)
    expect(rightArea.is('div')).toBeTruthy()
    expect(rightArea.children()).toHaveLength(2)

    const username = rightArea.childAt(0)
    expect(username.is('span')).toBeTruthy()

    const logLink = rightArea.childAt(1)
    expect(logLink.is(Link)).toBeTruthy()
  })

  test('Should link the brand to \'/\' if no brand link is given', () => {
    const newProps = clone(props)
    newProps.brandLink = undefined
    const wrapper = shallow(<AppBar {...newProps} />)

    const brand = wrapper.childAt(0)
    expect(brand.prop('to')).toBe('/')
  })

  test('Should link the brand to the given prop', () => {
    const brand = wrapper.childAt(0)
    expect(brand.prop('to')).toBe(props.brandLink)
  })

  test('Should display the brand name', () => {
    const brand = wrapper.childAt(0)
    const brandName = brand.childAt(0)

    expect(brandName.text()).toBe(props.brandName)
  })

  test('Should not display the username if there is none given', () => {
    const newProps = clone(props)
    newProps.userName = undefined
    const wrapper = shallow(<AppBar {...newProps} />)

    const rightArea = wrapper.childAt(1)

    // If the username does not display, there is only child (login / logout)
    expect(rightArea.children).toHaveLength(1)
  })

  test('Should display the username', () => {
    const rightArea = wrapper.childAt(1)
    expect(rightArea.children()).toHaveLength(2)

    const userName = rightArea.childAt(0)
    expect(userName.text()).toBe(props.userName)
  })

  test('Should display the login link if no username is given', () => {
    const newProps = clone(props)
    newProps.userName = undefined
    const wrapper = shallow(<AppBar {...newProps} />)

    const rightArea = wrapper.childAt(1)
    const logLink = rightArea.childAt(0)  // first el if there is no username

    expect(logLink.is(Link)).toBeTruthy()
    expect(logLink.prop('to')).toBe(props.loginLink)
    expect(logLink.children()).toHaveLength(1)

    const logLinkText = logLink.childAt(0)
    expect(logLinkText.text()).toBe('login')
  })

  test('Should display the logout link if a username is found', () => {
    const rightArea = wrapper.childAt(1)
    const logLink = rightArea.childAt(1) // 2nd el if there is a username

    expect(logLink.is(Link)).toBeTruthy()
    expect(logLink.prop('to')).toBe(props.logoutLink)
    expect(logLink.children()).toHaveLength(1)

    const logLinkText = logLink.childAt(0)
    expect(logLinkText.text()).toBe('logout')
  })
})
