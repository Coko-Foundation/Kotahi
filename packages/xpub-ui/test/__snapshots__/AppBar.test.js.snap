// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar Snapshot 1`] = `
<div
  className="root"
>
  <a
    className="link logo"
    href="some link"
    onClick={[Function]}
  >
    some brand
  </a>
  <div>
    <span
      className="item"
    >
      <span
        className="root"
      >
        <svg
          height={16}
          viewBox="0 0 24 24"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
            fill="none"
            stroke="black"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          />
          <circle
            cx="12"
            cy="7"
            fill="none"
            r="4"
            stroke="black"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          />
        </svg>
      </span>
      <span
        className="username"
      >
        some name
      </span>
    </span>
    <a
      className="item link"
      href="logout link"
      onClick={[Function]}
    >
      logout
    </a>
  </div>
</div>
`;
